modules/ocf_mail/files/spam/logging/log-mail

```diff
@@ -61,16 +92,35 @@ def parse_received_for_auth(header):
         }
 
 
-def clean_addr(addr):
+def parse_received_for_recipient(header):
+    """Attempt to parse the receipient address from the Received header.
+
+    The To header is required, but can sometimes be something like "To:
+    undisclosed-recipients:;", which is not helpful at all and otherwise
+    produces an empty output in the log ([null, null]). We'd prefer to have an
+    actual address shown, so we search through the Received headers for an
+    address to log.
+    """
+    match = re.search(
+        r'^\tfor <(.+?)>;(.*)$',
+        header,
+        re.MULTILINE,
+    )
+    if match:
+        # The recipient's name will be empty since only the email is parsed out
+        # of the Received header
+        return ('', match.group(1))
+
+
+def clean_addr(realname, addr):
     """Cleans up email address, attempting to normalize as much as possible.
 
     We keep the realname in case it says something useful (e.g. 'Cron Daemon').
 
-    >>> clean_addr('Chris Kuehl <cKuEhL@OCF.Berkeley.EDU>')
+    >>> clean_addr(('Chris Kuehl', '<cKuEhL@OCF.Berkeley.EDU>'))
```

Comment:
is this accurate? seems like it takes 2 arguments, not one?

Reply:
Yeah, I forgot a `*` here (or to separate the arguments, that would be clearer), good catch.